pilot/pkg/model/context.go

```diff
@@ -174,6 +177,33 @@ func (node *Proxy) GetRouterMode() RouterMode {
 	return StandardRouter
 }
 
+// SetSidecarScope identifies the sidecar scope object associated with this
+// proxy and updates the proxy Node. This is a convenience hack so that
+// callers can simply call push.Services(node) while the implementation of
+// push.Services can return the set of services from the proxyNode's
+// sidecar scope or from the push context's set of global services. Similar
+// logic applies to push.VirtualServices and push.DestinationRule. The
+// short cut here is useful only for CDS and parts of RDS generation code.
+//
+// Listener generation code will still use the SidecarScope object directly
+// as it needs the set of services for each listener port.
+func (node *Proxy) SetSidecarScope(ps *PushContext) {
+	instances := node.ServiceInstances
+	node.SidecarScope = ps.getSidecarScope(node, instances)
+}
+
+func (node *Proxy) SetServiceInstances(ps *PushContext) error {
```

Comment:
If what you need as an argument is `*Environment` why not have it?
Also can `ps` be nil here?

Reply:
nope..